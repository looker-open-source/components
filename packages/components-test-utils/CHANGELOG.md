# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.10.4](https://github.com/looker-open-source/components/compare/v0.10.3...v0.10.4) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.3](https://github.com/looker-open-source/components/compare/v0.10.2...v0.10.3) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.2](https://github.com/looker-open-source/components/compare/v0.10.1...v0.10.2) (2021-01-22)


### Bug Fixes

* On build properly set package.json "files" key ([#1806](https://github.com/looker-open-source/components/issues/1806)) ([c81c4bb](https://github.com/looker-open-source/components/commit/c81c4bb625c58ede49957aad90ac8d9d7b2c4b79))





## [0.10.1](https://github.com/looker-open-source/components/compare/v0.10.0...v0.10.1) (2021-01-22)


### Bug Fixes

* Properly build ems/cjs/ts for easy deep-imports ([#1804](https://github.com/looker-open-source/components/issues/1804)) ([2a523cd](https://github.com/looker-open-source/components/commit/2a523cd70b079944376c6dc87c18202e05a97b01))





# [0.10.0](https://github.com/looker-open-source/components/compare/v0.9.29...v0.10.0) (2021-01-19)


### Bug Fixes

* Properly configure build for tree-shaking support ([9202b72](https://github.com/looker-open-source/components/commit/9202b72bc5fb99eb3e1af7d9e56f3dc15b1df2a4))





## [0.9.15]

### Changed

- `@looker/components-test-utils` helpers will no longer produce `globalStyles`
