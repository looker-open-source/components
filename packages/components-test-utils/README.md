# Test Utilities

Shared functions used for mocking and testing Looker UI Components.
