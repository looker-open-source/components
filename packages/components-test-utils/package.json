{
  "name": "@looker/components-test-utils",
  "license": "MIT",
  "version": "0.9.12",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/looker-open-source/components",
    "directory": "packages/components-test-utils"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@looker/components-providers": "^0.9.12",
    "@testing-library/react": "^10.4.8",
    "@types/enzyme": "^3.10.5",
    "@types/jest": "^26.0.10",
    "@types/react": "^16.9.49",
    "@types/react-test-renderer": "^16.9.3",
    "enzyme": "^3.11.0",
    "jest-styled-components": "^6.3.4",
    "react": "^16.13.1",
    "react-test-renderer": "^16.13.1",
    "styled-components": "^4.4.1"
  },
  "peerDependencies": {
    "react": "^16.11"
  },
  "gitHead": "62febc162e3e45124f403e2c84fc741f68fe6714"
}
