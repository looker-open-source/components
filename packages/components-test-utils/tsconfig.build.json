{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./lib"
  },
  "include": ["../../types", "src/**/*"]
}
