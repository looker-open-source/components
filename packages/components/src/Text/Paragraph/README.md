# Paragraph

Renders a block level `<Paragraph />` element on the page.

```tsx
function Basic() {
  return <Paragraph>Paragraph Text</Paragraph>;
}
```

By default, paragraphs inherit the font size of the parent element.

## Truncate

At times you may want your `<Paragraph />` to truncate instead of the text wrapping, the `truncate` property will do that for you. For a multiline truncate, use `truncateLines`.

```tsx
function Truncate() {
  return (
    <Paragraph truncate={true}>
      This is a really long title that will need to truncate. It's gotta get
      real long so it hits the edge of the jest-dom virtual window size so I'm
      going to just keep typing and typing to make sure it triggers overflow as
      needed to prove that this is work properly. Are we there yet? Maybe? I
      sure hope so!
    </Paragraph>
  );
}
```

```tsx
function MultilineTruncate() {
  return (
    <Paragraph truncateLines={2}>
      This is a really long title that will need to truncate. It's gotta get
      real long so it hits the edge of the jest-dom virtual window size so I'm
      going to just keep typing and typing to make sure it triggers overflow as
      needed to prove that this is work properly. Are we there yet? Maybe? I
      sure hope so!
    </Paragraph>
  );
}
```

## Text Alignment

You can use the `textAlign` property to change the alignment of the rendered text, below is an example with aligned text put into `<Card />`s.

```tsx
function TextAlign() {
  return (
    <>
      <Card>
        <CardContent>
          <Paragraph> I am aligned left by default</Paragraph>
        </CardContent>
      </Card>
      <Card>
        <CardContent>
          <Paragraph textAlign="center">
            This is how you can center align Paragraph text
          </Paragraph>
        </CardContent>
      </Card>
      <Card>
        <CardContent>
          <Paragraph textAlign="right">
            This is how you can right align Paragraph text
          </Paragraph>
        </CardContent>
      </Card>
    </>
  );
}
```
