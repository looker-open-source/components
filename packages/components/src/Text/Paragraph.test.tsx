/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React from 'react'
import { createWithTheme } from '@looker/components-test-utils'
import { Paragraph } from './Paragraph'

test('A default Paragraph component', () => {
  const component = createWithTheme(<Paragraph>Hello</Paragraph>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('A Paragraph component resized', () => {
  const component = createWithTheme(
    <Paragraph fontSize="xxxxlarge">Hello</Paragraph>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('A Paragraph component aligned', () => {
  const component = createWithTheme(
    <Paragraph textAlign="right">Hello</Paragraph>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('A Paragraph component weight', () => {
  const component = createWithTheme(
    <Paragraph fontWeight="bold">Hello</Paragraph>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('A Paragraph component with variant', () => {
  const component = createWithTheme(
    <Paragraph variant="critical">Hello</Paragraph>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('A Paragraph component text transformed', () => {
  const component = createWithTheme(
    <Paragraph textTransform="uppercase">Hello</Paragraph>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('A Paragraph component wrapped', () => {
  const component = createWithTheme(<Paragraph wrap>Hello</Paragraph>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('A Paragraph component decorated', () => {
  const component = createWithTheme(
    <Paragraph textDecoration="line-through">Hello</Paragraph>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('A Paragraph component truncated', () => {
  const component = createWithTheme(<Paragraph truncate>Hello</Paragraph>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('A Paragraph component with multiline truncate', () => {
  const component = createWithTheme(
    <Paragraph truncateLines={2}>Hello</Paragraph>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
