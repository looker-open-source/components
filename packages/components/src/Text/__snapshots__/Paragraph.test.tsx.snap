// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Paragraph component aligned 1`] = `
.c0 {
  text-align: right;
  font-size: 1rem;
  line-height: 1.5rem;
}

<p
  className="c0"
  fontSize="medium"
>
  Hello
</p>
`;

exports[`A Paragraph component decorated 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5rem;
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

<p
  className="c0"
  fontSize="medium"
  textDecoration="line-through"
>
  Hello
</p>
`;

exports[`A Paragraph component resized 1`] = `
.c0 {
  font-size: 2.25rem;
  line-height: 2.75rem;
}

<p
  className="c0"
  fontSize="xxxxlarge"
>
  Hello
</p>
`;

exports[`A Paragraph component text transformed 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5rem;
  text-transform: uppercase;
}

<p
  className="c0"
  fontSize="medium"
>
  Hello
</p>
`;

exports[`A Paragraph component truncated 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<p
  className="c0"
  fontSize="medium"
>
  Hello
</p>
`;

exports[`A Paragraph component weight 1`] = `
.c0 {
  font-weight: 700;
  font-size: 1rem;
  line-height: 1.5rem;
}

<p
  className="c0"
  fontSize="medium"
  fontWeight="bold"
>
  Hello
</p>
`;

exports[`A Paragraph component with multiline truncate 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5rem;
  -webkit-box-orient: vertical;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  overflow: hidden;
}

<p
  className="c0"
  fontSize="medium"
>
  Hello
</p>
`;

exports[`A Paragraph component with variant 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5rem;
  color: #CC1F36;
}

<p
  className="c0"
  fontSize="medium"
>
  Hello
</p>
`;

exports[`A Paragraph component wrapped 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5rem;
  overflow-wrap: break-word;
}

<p
  className="c0"
  fontSize="medium"
>
  Hello
</p>
`;

exports[`A default Paragraph component 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5rem;
}

<p
  className="c0"
  fontSize="medium"
>
  Hello
</p>
`;
